app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -46,7 +53,10 @@ class DetailedItemFragment : Fragment() {
             binding.detailedItemImage.setImageURI(Uri.parse(it))
         }
 
+        initiateWishlistButton(binding)
+
         binding.shareButton.setOnClickListener { shareItem() }
+        binding.addToWishlist.setOnClickListener { updateWishlist(binding) }
```

Comment:
Can't you put all wishlist button related code in the initiateWishlistButton method?